### ISSUE TITLE
[Question] How can I get the value of a column when a row is pressed?

### ISSUE THREAD

**Issue by ginestopo** [thingsboard/thingsboard#7484]
**Component**

  * Generic

**Description**

![Captura de pantalla 2022-10-25 140049](https://user-images.githubusercontent.com/12192256/197768032-81f956ea-7cf5-4de8-af22-1ea8f1ff9c35.png)

In our application we have several receptors (in this case receptor 1 and receptor 2)
I would like to show an alert window displaying the battery value of the selected receptor when I press a row in a entities table widget. To do that, I tried writing a javascript file included in a custom action inside the widget which is shown below.
Nevertheless it outputs a json which does not contain the value of the battery, hence I cannot print it in the alert window.

I wonder if there is a way to do that.

I will take advantage of the situation to ask too: Is there a way of changing the value shown by an analog display widget depending on the selected row mention above?

Thank you very much in advance! Have a nice day 

```
var title;
var content;
if (entityName) {
  title = entityName + ' details';
  content = '<b>Entity name</b>: ' + entityName;
  if (additionalParams && additionalParams.entity) {
    var entity = additionalParams.entity;
    if (entity.id) {
      content += '<br><b>Entity type</b>: ' + entity.id.entityType;
    }
    if (!isNaN(entity.temperature) && entity.temperature !== '') {
      content += '<br><b>Temperature</b>: ' + entity.temperature + ' °C';
    }

    //content for getting the battery value of the current row
    content += '<br><b>Battery</b>: ' + JSON.stringify(widgetContext.data) + ' %';

    
  }
} else {
  title = 'No entity information available';
  content = '<b>No entity information available</b>';
}

showAlertDialog(title, content);

function showAlertDialog(title, content) {
  setTimeout(function() {
    widgetContext.dialogs.alert(title, content).subscribe();
  }, 100);
}

```

**Environment**

 * OS:  Windows.
 * ThingsBoard: v3.4.1 Thingsboard Demo
 * Browser: Firefox 105.0.3 (64-bit)

**Comment by devaskim** [comment]
```
let entity = additionalParams.entity;
let dialogs = widgetContext.$scope.$injector.get(widgetContext.servicesMap.get('dialogs'));
let title = `My consumption`;
let content = `${entity.Consumption} kW/h`;
dialogs.alert(title, content).subscribe(
    result => {

    }
);
```

**Comment by devaskim** [comment]
To update gauge you need 3 things:
- update dashboard state in custom action triggered by clicking on table's row
```
    let state = widgetContext.stateController.getStateParams();
    // Field's name "selectedEntity" may be any
    state.selectedEntity = {
        entityId: {
            id: entityId,
            entityType: 'DEVICE'
        },
        entityName: entityName,
        entityLabel: entityLabel
    };
    widgetContext.stateController.updateState(state);
```
- use alias called `Entity from dashboard state` with option `State entity parameter name` set to `selectedEntity ` (see comment above about naming)
- use that alias for gauge widget

**Comment by ginestopo** [comment]
@devaskim Thanks a lot! I will give it a try soon and let you know the results. Have a nice day 😄

**Comment by devaskim** [comment]
Hi @ginestopo 
I didn't test if table widget supports multiple `On row click` actions. So I suggested you to combine all code in one `Custom action`.

If support is, you dont' need to update dashboard state manually as I described in [my previous comment](https://github.com/thingsboard/thingsboard/issues/7484#issuecomment-1292132637). Just create another `On row click` action with type `Update current dashboard state`

Have a nice day too)))
